tavern/cli/tomeuploader/main.go

```diff
@@ -0,0 +1,308 @@
+	escapedEldritchContent := strings.ReplaceAll(eldritchContent, "\\", "\\\\")
+	escapedEldritchContent = strings.ReplaceAll(escapedEldritchContent, "\"", "\\\"")
+	escapedEldritchContent = strings.ReplaceAll(escapedEldritchContent, "\n", "\\n")
+
+	// Create files first and collect their IDs
+	// fileIDs := make([]string, 0)
+	// for _, file := range files {
+	// 	mutation := fmt.Sprintf(`
+	//         mutation {
+	//             createFile(name: "%s", content: "%s") {
+	//                 id
+	//             }
+	//         }
+	//     `, file.Name, base64.StdEncoding.EncodeToString(file.Content))
+
+	// 	resp, err := sendGraphQLRequest(endpoint, mutation)
+	// 	if err != nil {
+	// 		return fmt.Errorf("failed to create file %q: %w", file.Name, err)
+	// 	}
+
+	// 	var result struct {
+	// 		Data struct {
+	// 			File struct {
+	// 				ID string `json:"id"`
+	// 			} `json:"createFile"`
+	// 		}
+	// 	}
+	// 	if err := json.Unmarshal(resp, &result); err != nil {
+	// 		return fmt.Errorf("failed to parse create file response: %w", err)
+	// 	}
+	// 	fileIDs = append(fileIDs, result.Data.File.ID)
+	// }
+
+	// Convert paramDefs to a JSON string
+	paramDefsJSON, err := json.Marshal(metadata.ParamDefs)
+	if err != nil {
+		return fmt.Errorf("failed to marshal paramDefs: %w", err)
+	}
+	paramDefsJSONString := string(paramDefsJSON)
+
+	// Escape double quotes for proper string embedding in GraphQL
+	paramDefsJSONString = strings.ReplaceAll(paramDefsJSONString, "\"", "\\\"")
+
+	// Construct the GraphQL mutation for creating a tome
+	createTomeMutation := fmt.Sprintf(`
+        mutation {
+            createTome(input: {
+                name: "%s",
+                description: "%s",
+                author: "%s",
+                supportModel: %s,
+                tactic: %s,
+                paramDefs: "%s",
+                eldritch: "%s"
+
+            }) {
+                id
+            }
+        }
+    `, metadata.Name, metadata.Description, metadata.Author, metadata.SupportModel, metadata.Tactic, paramDefsJSONString, escapedEldritchContent)
+	fmt.Println(createTomeMutation)
+	// Send the createTome mutation
+	_, err = sendGraphQLRequest(endpoint, createTomeMutation, cookie)
+	if err != nil {
+		return fmt.Errorf("failed to create tome %q: %w", metadata.Name, err)
+	}
+
+	return nil
+}
+
+func sendGraphQLRequest(endpoint string, query string, cookie string) ([]byte, error) {
+	// Marshal the query into a JSON request body
+	requestBody, err := json.Marshal(map[string]string{
+		"query": query,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request body: %w", err)
+	}
+
+	// Create a new HTTP request with the necessary headers
+	req, err := http.NewRequest("POST", endpoint, bytes.NewBuffer(requestBody))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	// Set the content type and cookie headers
+	req.Header.Set("Content-Type", "application/json")
```

Comment:
Just use token.Authenticate instead, have users pass in the token via env, and if none is set and a new token is obtained just save it to a file for now (and try reading the value if no env is set)